javascript/URDFLoader.js

```diff
@@ -515,35 +547,27 @@ class URDFLoader {
                 rpy = this._processTuple(n.getAttribute('rpy'));
 
             } else if (type === 'material') {
+                const materialName = n.getAttribute('name');
+                if (n.children.length) {
+                    this.forEach(n.children, c => {
+                        switch (c.nodeName.toLowerCase()) {
+
+                            case 'color':
+                                Object.assign(material, this._processMaterial('color', c.getAttribute('rgba')));
+                                break;
+                            case 'texture':
+                                if (materialName in materialMap) {
+                                    Object.assign(material, materialMap[materialName]);
+                                } else {
+                                    Object.assign(material, this._processMaterial('texture', c.getAttribute('filename'), packages, path));
+                                }
 
-                this.forEach(n.children, c => {
-
-                    if (c.nodeName.toLowerCase() === 'color') {
-
-                        const rgba = c.getAttribute('rgba')
-                            .split(/\s/g)
-                            .map(v => parseFloat(v));
-
-                        material.color.r = rgba[0];
-                        material.color.g = rgba[1];
-                        material.color.b = rgba[2];
-                        material.opacity = rgba[3];
-
-                        if (material.opacity < 1) material.transparent = true;
-
-                    } else if (c.nodeName.toLowerCase() === 'texture') {
-
-                        const filename = c.getAttribute('filename').replace(/^(package:\/\/)/, '');
-                        const filePath = this._resolvePackagePath(packages, filename, path);
-
-                        material.map = this._textureloader.load(filePath);
-
-                    }
-
-                });
-
+                        }
+                    });
+                } else {
+                    Object.assign(material, materialMap[materialName]);
```

Comment:
Using `Object.assign` here won't work well for fields like `color` because then every instance of the materials used here will have the same instance of the `THREE.Color` object. Using `material.color.copy(materialMap[materialName])` seems like the better way to handle the color copy.

It may be tedious but it might be necessary to handle every field assignment individually.